eem-sparks/src/main/java/com/nice/eem/sparkathon/UploadFileToS3.java

```diff
@@ -0,0 +1,55 @@
+package com.nice.eem.sparkathon;
+
+import com.amazonaws.auth.AWSCredentials;
+import com.amazonaws.auth.AWSStaticCredentialsProvider;
+import com.amazonaws.auth.BasicAWSCredentials;
+import com.amazonaws.auth.InstanceProfileCredentialsProvider;
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.AmazonS3ClientBuilder;
+
+public class UploadFileToS3 {
+
+
+    //uploading file to the given s3 bucket using client created by the builder
+    public static void uploadFileToS3(AmazonS3 s3Client, String bucketName, String fileName, String filePath) {
+        s3Client.putObject(bucketName, fileName, filePath);
+        System.out.println("File uploaded successfully");
+    }
+
+    //downloading file from s3 bucket using client created by the builder
+    public static void downloadFileFromS3(AmazonS3 s3Client, String bucketName, String fileName, String filePath) {
+        s3Client.getObject(bucketName, fileName);
+        System.out.println("File downloaded successfully");
+    }
+
+    /**
+     *
+     * @param iscloudDeployment
+     * @return
+     */
+    public AmazonS3 buildS3Client(boolean iscloudDeployment) {
+        AWSCredentials awsCredentials;
+        AmazonS3 s3Client;
+        String awsRegion = "us-east-1";
+        if (iscloudDeployment) {
+
+            String accessKey = "xxxxxxxxxxxxxxxxxx";
+            String secretKey = "xxxxxxxxxxxxxxxxxx";
+            awsCredentials = new BasicAWSCredentials(accessKey, secretKey);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

If possible, use temporary security credentials (IAM roles) instead of long-term access keys.
Long-term access keys, such as those associated with IAM users and AWS account root users, remain valid until you manually revoke them. However, temporary security credentials obtained through IAM roles and other features of the AWS Security Token Service expire after a short period of time. Use temporary security credentials to help reduce your risk in case credentials are accidentally exposed.
Learn more about [best practices for managing AWS access keys](https://docs.aws.amazon.com/general/latest/gr/aws-access-keys-best-practices.html).